stripe/http_client.py

```diff
@@ -114,12 +114,20 @@ def request(self, method, url, headers, post_data=None):
 
         try:
             try:
-                result = requests.request(method,
-                                          url,
-                                          headers=headers,
-                                          data=post_data,
-                                          timeout=self._timeout,
-                                          **kwargs)
+                if(session is not None):
+                    result = session.request(method,
```

Comment:
Rather than have two branches that are largely identical, we should probably refactor a bit. Something like:

```py
requestor = session if session is not None else requests
requestor.request(method, ...
```